### PR TITLE
Provider validate credentials

### PR DESCRIPTION
**Pull request by kbrock**
We hashed around some ideas for provider's validate_credentials

- Wanted to get Compute and Baremetal into there
- Wanted the logic for each of those to be specific
- I didn't touch the auth_type.nil? case - we can discuss.
- I didn't handle cinder or swift managers. Anything special there?
- I did not see an issue with `ActiveSupport::Concerns`, but it is wrong and was bugging me

### CODE REVIEW

File: app/models/manageiq/providers/openstack/cloud_manager.rb [hunk 1]
```diff
@@ -503,6 +503,17 @@ def swift_service
...
 
+  def verify_credentials(auth_type = nil, options = {})
+    options[:service] ||= "Compute"
+    ret = super
+    return ret unless auth_type.nil?
+
+    capabilities["events"] = !!event_monitor_available?
+    save! if changed?
+
+    true
```
Comment (kbrock) [inline app/models/manageiq/providers/openstack/cloud_manager.rb:514]: this is where the manager_mixin `else` went

File: app/models/manageiq/providers/openstack/infra_manager/host/operations.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module ManageIQ::Providers::Openstack::InfraManager::Host::Operations
-  include ActiveSupport::Concern
+  extend ActiveSupport::Concern
```
Comment (kbrock) [inline app/models/manageiq/providers/openstack/infra_manager/host/operations.rb:2]: this is a tangent.
I think the existing one is a bug but can pull into another PR if you want

File: app/models/manageiq/providers/openstack/manager_mixin.rb [hunk 3]
```diff
@@ -318,18 +317,16 @@ def verify_event_monitor(_options = {})
   end
 
   def verify_credentials(auth_type = nil, options = {})
-    case auth_type
-    when :default
-      verify_api_credentials
-    when :amqp
-      verify_amqp_credentials
-    else
```
Comment (kbrock) [inline app/models/manageiq/providers/openstack/manager_mixin.rb:326]: I'm pretty sure the `else` is for a `nil` and not for a different `auth_type`.

**Review by agrare — APPROVED** [review]
